src/Platform/Vulkan/VulkanUniformMetaData.hh

```diff
@@ -0,0 +1,69 @@
+#ifndef PLATFORM_VULKAN_RENDER_API_VULKAN_UNIFORM_HH
+#define PLATFORM_VULKAN_RENDER_API_VULKAN_UNIFORM_HH
+
+// Render API
+#include "Core/RenderAPI/EspUniformMetaData.hh"
+
+// libs
+#include "volk.h"
+
+namespace esp
+{
+  struct VulkanMetaUniform
+  {
+    EspUniformShaderStage m_stage;
+
+    // Size of basic data type.
+    uint32_t m_size_of_data_chunk;
+
+    // Number of textures or objects for uniform.
+    uint32_t m_count_of_elements;
+
+    uint32_t m_binding;
+
+    EspUniformType m_uniform_type;
+
+    VulkanMetaUniform(EspUniformShaderStage stage,
+                      uint32_t size_of_data_chunk,
+                      uint32_t count_of_elements,
+                      uint32_t binding,
+                      EspUniformType uniform_type);
+  };
+
+  struct VulkanMetaDescriptorSet
+  {
+    std::vector<VulkanMetaUniform> m_meta_uniforms;
+    inline void push_back(VulkanMetaUniform& uniform_data) { m_meta_uniforms.push_back(uniform_data); }
+  };
+
+  struct VulkanUniformMetaData : public EspUniformMetaData
+  {
+   private:
+    inline void push_back_to_current(VulkanMetaUniform& uniform_data)
```

Comment:
@DawidSzpejna are you sure this method is correct? Shoudn't it be like
"   m_meta_descriptor_sets.back().M_META_UNIFORMS.push_back(uniform_data);   "

Reply:
He He He. I would appreciate if you read the code of your colleague more attentively. The function `push_back()` is implemented for `VulkanMetaDescriptorSet` too :D